### DISCUSSION TITLE
Error Message for Xcode Default App Examples

### DISCUSSION THREAD

**Discussion by giiyms** [sweetpad-dev/sweetpad#64] [Any discussions (ideas, questions, ...)]
Hello,

I have just created a basic watch app with ios companion app but vscode gives me this weird error:
<img width="1039" alt="image" src="https://github.com/user-attachments/assets/37e431c3-768c-412b-bf33-95e4b9b727c2">

'main' attribute cannot be used in a module that contains top-level code

The project builds fine in Xcode and using build commands but the errors are annoying.

I tried adding "Pass '-parse-as-library' to compiler invocation if this is intentional" but no success. Any ideas?

**Comment by hyzyla** [comment]
Hello! I just tried Googling your problem and found this issue with the Swift compiler: https://github.com/swiftlang/swift/issues/55127. Could you check if any of the workarounds from this discussion would work for you?

It's also worth mentioning that this "problems" are parsed from the build logs, so if you change anything, you’ll need to rebuild the project to test it

**Reply by giiyms** [reply]
Hi thanks for the response and link.

Do you perhaps know how I can add the:

-XSwiftc -parse-as-library

To the sweetpad build command?

**Reply by hyzyla** [reply]
try to add this to your ".vscode/settings.json" config file:

```jsonc
{
  "sweetpad.build.args": ["OTHER_SWIFT_FLAGS=\"-parse-as-library\""],
}
````

![CleanShot 2024-11-03 at 12 42 20@2x](https://github.com/user-attachments/assets/5abbf4d5-bf59-47f9-9361-c032dc673891)

**Reply by giiyms** [reply]
thank you it worked!
